src/src/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -272,6 +273,8 @@ else if (response.containsKey(AuthenticationConstants.OAuth2.ACCESS_TOKEN)) {
             
             //Set family client id on authentication result for TokenCacheItem to pick up
             result.setFamilyClientId(familyClientId);
+        } else {
+            result = null;
```

Comment:
This actually should throw an AuthenticationException, but I will submit it in next PR